### PR TITLE
Space Ninjas

### PR DESCRIPTION
**Pull request by Kyep**
This PR fixes up Space Ninjas to the point that can function as an event antag.

Changes:

1) Rigsuit changes:
- Ninja is now invisible to sec/medical/etc huds while cloaked. (FIXES major bug where people with certain HUDs could still see them, a bug that made ninjas pretty much unplayable)
- Ninja's hardsuit's shuriken projector now works. Middle-click with the blade module selected to use it. The shurikens do 40 BURN damage, and have a stun effect. BUT they have a long (10s+) recharge time, they drain your suit's power quickly, and they cannot be spammed. Side effect: FIXED major bug where many energy guns on rigsuits don't work, but DO cripple your power supply on use.
- Ninja's suit power cell now starts as a super-cap cell, not a normal cell.
- Ninjas no longer get the fabricator rigsuit module. It created tiles for throwing, but it was useless.
- Ninjas also no longer get an AI container module. There's no reason they need to carry an AI around in their suit.
- Ninjas now get a jetpack module for their rigsuit. Obviously, SPACE ninjas should be at home in space. This is in preparation for them possibly later being able to spawn in space.

2) Loadout creation:
- 'ninja' is now a loadout option in the admin -> select equipment verb.
- Since they can't use backpacks, most of their gear is stored in a storage implant. This has the side effect of making it much harder to recover their gear if they are killed.

3) Weapons:
- Hardsuit grenades: Knockout N2O clusterbomb, 1x Spider-spawner (spiders are NOT hostile to ninja of the Spider Clan, you can use these to protect your recharge point), Cluster Smokebomb "Ninja Vanish".
- Hardsuit weapon: energy katana (can emag doors, good melee weapon too. Can be replaced if you drop it.)
- Hardsuit weapon: energy shuriken (massive nano-shuriken, does lots of damage but takes ages to recharge and is very costly, energy wise)

4) Martial Arts:
- Ninjas now use Ninjitsu, the proper ninja martial art for ninjas. No more Sleeping Carp!
- Right now, Ninjitsu is a midway point between nothing and Sleeping Carp. It only gives 75% deflect, and has almost no special moves. Still, it doesn't prohibit the use of ranged weapons, either. 
- Ninjitsu is intended to help the ninja stave off the odd taser shot while fleeing, while at the same time not making it impossible to hit them if the ninja gets themselves trapped in a corridor, and the crew fill it with laser fire.

### CODE REVIEW

**Comment by FreeStylaLT** [comment]
@DaveTheHeadcrab

**Comment by tigercat2000** [comment]
Rigsuits aren't ready for player usage. They have some serious issues with their functionality, and can outright completely break under certain circumstances.

**Comment by MarsM0nd** [comment]
Shooting mounted guns doesn't use any charge, and not sure how correct that is, but i think ```gun.process_fire(target,holder.wearer)``` can just be replaced by ```gun.afterattack(target,holder.wearer)```, which handles the chambering and all that, it works at least, and i figured that guns are coded in a way that it first handles the meele attack, and then the afterattack handles the shooting.

**Comment by Fox-McCloud** [comment]
Anything that remotely touches or uses rigsuits isn't really viable right now and shouldn't really be used by players--to say the least `WIP` is applicable to them.

I had planned on implementing TG's ninja's (which are a dramatic OOP improvement over their previous iteration while retaining the same mechanics) after the food rework; it'll be much more gracefully handled than anything the RIGsuits currently have.

**Comment by Allfd** [comment]
While having issues as a game mode, rig suits will never be fixed if we don't start merging in fixes.

**Comment by tigercat2000** [comment]
@Allfd this doesn't fix anything about rigsuits, however- if it did, it wouldn't be an issue

**Comment by Kyep** [comment]
@tigercat2000 correct. Hence why this is an event or admin antag, and the only way to create one is select equipment - normal players can't become one.

@MarsM0nd prior to this PR, they both fail to fire AND use charge. This fixes them so they use charge if and only if they actually fire (and, they can fire). 

@Fox-McCloud How certain are you that you will implement the TG port?

@Allfd That's pretty much the idea.

**Comment by Fox-McCloud** [comment]
>How certain are you that you will implement the TG port?

Very certain; I wanted to submit the food PR during the freeze (but obviously couldn't) and also start work on ninjas, again during the freeze---as soon as the food PR is merged in, I'll be starting work on the Ninja PR--can't say it'll be up in 2 days within it happening, but it's something I will be actively working on.

**Comment by Kyep** [comment]
In that case, I think I'll close this. A port of TG ninjas would probably be better than this.

**Comment by Spacemanspark** [comment]
Do /tg/ ninjas still have the ability to send you to the neat little prison thing, @Fox-McCloud?

**Comment by Fox-McCloud** [comment]
@Spacemanspark 

Not really sure; it's been a longggg time since I delved into ninja code in depth; I've just glanced at it.

**Comment by Kyep** [comment]
Reopening this, because:

1) Other rigsuit fix PRs have been made, and merged, e.g: https://github.com/ParadiseSS13/Paradise/pull/6535 - given that we're willing to merge those, I see no reason not to merge this, as most of the point of this is fixing horribly broken things with ninja rigsuits (weapon not working, cloak not working, etc). If we are willing to merge other rigsuit fixes, why not this? It is primarily a series of rigsuit fixes. 

2) This ninja rigsuit is admin-only. I keep seeing admins try (and fail) to create working ninjas by spawning gear, dressing themselves, etc. There's demand for an adminbus ninja. Even if TG ninjas are later added, this could still see use as a more powerful adminbus event ninja. There is demand for it, and it isn't obsoleted by any future TG ninja port. So, why not? It cannot hurt.

**Comment by Fox-McCloud** [comment]
I still have every intention of implementing TG's reworked Ninja's, which is still wayyy ahead of this in terms of functionality--it's something admins can actually use and is more suited for gameplay than this version is.

Yes, rigsuits have been fixed up, but they're still nowhere near a "ready" state in terms of functionality---there's still lots of problems present and quirks about them.

**Comment by Kyep** [comment]
@Fox-McCloud Yes, there are still issues with rigsuits that make them unsuitable for player usage. Thing is, they'll never get better unless we merge PRs like this that improve them. Even assuming TG ninjas are ported later, this still has some usefulness as a stronger, admin-only event ninja. There's no reason not to merge it. It makes some progress fixing rigsuit issues, and it can't hurt.

**Comment by Fox-McCloud** [comment]
There's no point in merging something that's inherently broken, poorly balanced, and redundant when there's a better option available to us--it's just going to cause confusion and perplexity when they have multiple space ninja options available to them.

Stating that we must merge in other systems to justify fixing the base system is completely illogical; the base systems should work, first, before expanded upon--not expanded upon to justify fixing them to begin with. We're willing to merge fix PRs to rigsuits because it makes the system a bit more tennable; it doesn't mean they're necessarily in a good spot to support more features, yet.

File: code/datums/outfits/outfit_admin.dm [hunk 1]
```diff
@@ -1064,3 +1064,55 @@
 	var/obj/item/weapon/card/id/I = H.wear_id
 	if(istype(I))
 		apply_to_card(I, H, get_all_accesses() + get_all_centcom_access(), name, "syndie")
+
+
+/datum/outfit/admin/chrono
+	name = "Chrono Legionnaire"
```
Comment (Crazylemon64) [inline code/datums/outfits/outfit_admin.dm]: Why is this duplicated

File: code/modules/martial_arts/martial.dm [hunk 2]
```diff
@@ -168,6 +168,30 @@
...
+	if(!istype(user) || !user)
+		return
+	learn_ninjitsu(user)
+
+/obj/item/weapon/ninjitsu_scroll/proc/learn_ninjitsu(mob/living/carbon/human/user as mob)
+	if(!istype(user) || !user)
+		return
+	to_chat(user, "<span class='sciradio'>You have learned the ancient martial art of the Space Ninja! \
+					Your hand-to-hand combat has become much more effective, and you are now able to deflect any projectiles directed toward you. \
+					However, you are also unable to use any ranged weaponry. \
```
Comment (Crazylemon64) [inline code/modules/martial_arts/martial.dm]: I'm pretty sure this is copy-pasta'd from the carp scroll.

**Comment by Kyep** [comment]
@Crazylemon64 Changed.

**Comment by TullyBurnalot** [comment]
After discussion with the Maintainers, while we can absolutely get behind the concept of Space Ninjas, the system this is built on is still very much WIP, unreliable, quirky and horridly unbalanced. As such, we don't think this would be a positive addition to the codebase _at this time_.

**Pull request closed without merging**